src/networking.c

```diff
@@ -3911,7 +4169,20 @@ void clientKillCommand(client *c) {
      * only after we queued the reply to its output buffers. */
     if (close_this_client) c->flag.close_after_reply = 1;
 client_kill_done:
-    zfree(client_filter.ids);
+    freeClientFilter(&client_filter);
+}
+
+static void freeClientFilter(clientFilter *filter) {
+    zfree(filter->ids);
+    sdsfree(filter->flags);
```

Comment:
To be on the safer side, could we perform NULL check for these as well to avoid double free?